src/libguac/guacamole/socket.h

```diff
@@ -136,11 +136,25 @@ void guac_socket_free(guac_socket* socket);
  * to ensure neither side of the socket times out while the socket is open.
  * This ping will take the form of a "nop" instruction.
  *
+ * @deprecated
+ *     Manually starting the keep alive process on sockets is no longer
+ *     necessary, as the sockets enable the "nop" ping by default.
+ * 
  * @param socket
  *     The guac_socket to declare as requiring an automatic keep-alive ping.
  */
 void guac_socket_require_keep_alive(guac_socket* socket);
 
+/**
+ * Enables the keep-alive ping functionality on the given socket, causing it
+ * to automatically send a "nop" instruction ping periodically while the
+ * socket is open.
+ * 
+ * @param socket
+ *     The guac_socket on which to enable the automatic keep-alive ping.
+ */
+void guac_socket_enable_keep_alive(guac_socket* socket);
```

Comment:
Why a new function rather than `guac_socket_require_keep_alive()`?

Reply:
I was thinking of just changing the name, since it isn't that we "require" keep alive on a particular socket, we're just enabling it across the board.

Reply:
Well ... internally, yes, I agree it's more of an "enable" than a "require".

The idea behind the original `guac_socket_require_keep_alive()` naming was to define that function as a means of declaring a requirement, rather than imposing a specific method for socket implementations to satisfy that requirement, decoupling the statement of need from how/whether the need is met.

Admittedly, the wording in the documentation of `guac_socket_require_keep_alive()` which explicitly specifies a "keep-alive ping" and states that "this ping will take the form of a "nop" instruction" probably defeats that intent.

Reply:
Okay, I just kept "require" and undid the deprecation of it for the time being.  We can change it, if that makes sense, but I'm also all for not making changes when there isn't a great reason to change anything.